indexing-service/src/main/java/org/apache/druid/indexing/common/task/IndexTask.java

```diff
@@ -1199,6 +1201,15 @@ public InputSource getInputSource()
       return inputSource;
     }
 
+    public InputSource getNonNullInputSource(TaskToolbox toolbox)
+    {
+      Preconditions.checkNotNull(inputSource, "inputSource");
+      if (inputSource instanceof TaskInputSource && toolbox != null) {
+        return ((TaskInputSource) inputSource).withTaskToolbox(toolbox);
+      }
+      return inputSource;
+    }
```

Comment:
I think we might need to create it lazily since we don't have the `TaskToolbox` while setting the `inputSource` in the constructor of `IndexIOConfig`.

We can avoid multiple calls to `withTaskToolbox` by using an `AtomicReference` and initializing it lazily but just once (same way that we have handled `ParallelIndexSupervisorTask.inputSource`.)

Reply:
Done